### PR TITLE
THORN-2112, update the 4.x jwt-auth code to use smallrye-jwt impl

### PR DESCRIPTION
**Pull request by starksm64**
- [ x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ x] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [ x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [ x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by starksm64** [comment]
This addresses #971

File: testsuite/tck/jwt-auth/pom.xml [hunk 1]
```diff
@@ -12,7 +12,11 @@
 
   <artifactId>thorntail-tck-jwt-auth</artifactId>
   <name>TCK: MicroProfile JWT Auth</name>
-  
+
+  <properties>
```
Comment (kenfinnigan) [inline testsuite/tck/jwt-auth/pom.xml:16]: Could this property be removed and just use `version.mp-jwt-auth`?

File: core/kernel/src/main/java/io/thorntail/config/impl/sources/ConfigSources.java [hunk 2]
```diff
@@ -3,7 +3,8 @@
 /**
  * Created by bob on 2/7/18.
  */
-class ConfigSources {
```
Comment (kenfinnigan) [inline core/kernel/src/main/java/io/thorntail/config/impl/sources/ConfigSources.java]: Not sure this change was intentional

File: testsuite/tck/jwt-auth/pom.xml (hunk 1 above)
Comment (starksm64) [reply testsuite/tck/jwt-auth/pom.xml:16]: I had tried that, but this property is not being picked up from the bom even though the parent pom is pulling it in:

```
[jwt-auth 907]$ git diff
diff --git a/testsuite/tck/jwt-auth/pom.xml b/testsuite/tck/jwt-auth/pom.xml
index baedccf63..bae1bb238 100644
--- a/testsuite/tck/jwt-auth/pom.xml
+++ b/testsuite/tck/jwt-auth/pom.xml
@@ -56,13 +56,13 @@
     <dependency>
       <groupId>org.eclipse.microprofile.jwt</groupId>
       <artifactId>microprofile-jwt-auth-tck</artifactId>
-      <version>${version.mp-jwt-auth.tck}</version>
+      <version>${version.mp-jwt-auth}</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.eclipse.microprofile.jwt</groupId>
       <artifactId>microprofile-jwt-auth-tck</artifactId>
-      <version>${version.mp-jwt-auth.tck}</version>
+      <version>${version.mp-jwt-auth}</version>
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
[jwt-auth 908]$ mvn clean compile
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck:jar must be a valid version but is '${version.mp-jwt-auth}'. @ io.thorntail.tck:thorntail-tck-jwt-auth:[unknown-version], /Users/starksm/Dev/JBoss/Microprofile/starksm64-thorntail/testsuite/tck/jwt-auth/pom.xml, line 59, column 16
[ERROR] 'dependencies.dependency.version' for org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck:test-jar must be a valid version but is '${version.mp-jwt-auth}'. @ io.thorntail.tck:thorntail-tck-jwt-auth:[unknown-version], /Users/starksm/Dev/JBoss/Microprofile/starksm64-thorntail/testsuite/tck/jwt-auth/pom.xml, line 65, column 16
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project io.thorntail.tck:thorntail-tck-jwt-auth:4.0.0-SNAPSHOT (/Users/starksm/Dev/JBoss/Microprofile/starksm64-thorntail/testsuite/tck/jwt-auth/pom.xml) has 2 errors
[ERROR]     'dependencies.dependency.version' for org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck:jar must be a valid version but is '${version.mp-jwt-auth}'. @ io.thorntail.tck:thorntail-tck-jwt-auth:[unknown-version], /Users/starksm/Dev/JBoss/Microprofile/starksm64-thorntail/testsuite/tck/jwt-auth/pom.xml, line 59, column 16
[ERROR]     'dependencies.dependency.version' for org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck:test-jar must be a valid version but is '${version.mp-jwt-auth}'. @ io.thorntail.tck:thorntail-tck-jwt-auth:[unknown-version], /Users/starksm/Dev/JBoss/Microprofile/starksm64-thorntail/testsuite/tck/jwt-auth/pom.xml, line 65, column 16
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

File: core/kernel/src/main/java/io/thorntail/config/impl/sources/ConfigSources.java (hunk 2 above)
Comment (starksm64) [reply core/kernel/src/main/java/io/thorntail/config/impl/sources/ConfigSources.java]: No, it has been reverted.

File: testsuite/tck/jwt-auth/pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply testsuite/tck/jwt-auth/pom.xml:16]: Very weird. Thanks for checking

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**